thirdparty/github.go

```diff
@@ -148,6 +148,9 @@ func GetGithubCommits(ctx context.Context, oauthToken, owner, repo, ref string,
 	commits, resp, err := client.Repositories.ListCommits(ctx, owner, repo, &options)
 	if resp != nil {
 		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
```

Comment:
I worry the status code could potentially be something else in the 200 series and not indicate an error (not a result of your changes).

Maybe an alternative is to parseGithubErrorResponse (assuming resp isn't nil) within the `if err != nil` block given `err != nil` implies the status code [isn't in the 200 series](https://github.com/google/go-github/blob/339ae9b3ec8be6eda90fa75a03bb1950b726d529/github/github.go#L785).

Reply:
I got this wrong. Only the inverse is true: if `err == nil` then the status code is in the 200 series.

Reply:
Good find! I've reorganized a bit